docs/architecture-decisions/0005-implement-an-accessible-navigation-system.md

```diff
@@ -0,0 +1,214 @@
+  some javascript to manipulate the DOM prior to instantiating the menu, it does
+  mean that using this library increases the maintenance burden of the theme.
+
+* The library is javascript-only, and is not distributed with any CSS rules.
+  While we can develop and maintain these rules ourselves, it makes this library
+  less of a turnkey solution.
+
+
+### Option 3: CSS-only menu libraries
+
+A third option would be to pursue a "pure CSS" navigation menu, which can be
+found described in [several](https://purecss.io/menus/) [places](https://devsnap.me/css-menu) around the web.
+
+* Pros: A navigation menu which uses only CSS styles is technically much
+  simpler, lacking the javascript that would need to be maintained (either by
+  ourselves or by a dependency maintainer).
+
+* Cons: [These types of navigation systems are likely not going to be fully
+  accessible](https://moderncss.dev/css-only-accessible-dropdown-navigation-menu/). There are elements of the WCAG standards which cannot be satisfied
+  by stylesheets alone. An example is [WCAG 1.4.13](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html), which requires that revealed
+  content like a submenu must be dismissable without changing the focus on a
+  page.
+
+
+### Option 4: A bespoke solution
+
+While attempting to implement the Accessible Menu library, we asked for feedback
+about our progress from Rich Caloggero, who was involved in our earlier
+accessibility testing. Rich helpfully assembled a [mockup](https://richcaloggero.github.io/tools/createNavigation.html) demonstrating a usable
+navigation system with very little javascript (and no stylesheets).
+
+#### Benefits of a bespoke approach
+
+The amount of javascript which is needed is potentially very, very small. The
+initial snippet which Rich provided was just this:
+
+```js
+$submenus = document.querySelectorAll("nav ul ul");
+$submenus.forEach(x => x.setAttribute("hidden", ""));
+$nav.removeEventListener("click", clickHandler);
+$nav.addEventListener("click", clickHandler);
+
+function clickHandler (e) {
+if (e.target instanceof HTMLButtonElement) {
+e.preventDefault();
+const menu = e.target.parentElement instanceof HTMLLIElement? e.target.nextElementSibling  : e.target.parentElement.nextElementSibling;
+menu.hidden = not(menu.hidden);
+e.target.setAttribute("aria-expanded", menu.hidden? "false" : "true");
+} // if
+} // clickHandler
+```
+
+If we choose HTML elements which bring native functionality (`button`, `a`, and
+header tags, for example) then we not only need less javascript but we are more
+likely to match functionality which users are already familiar with using. The
+only conceptual need we are solving is how to expand and collapse a submenu -
+everything else is already present.
+
+The styling rules required for this approach should likewise be minimal.
+
+#### Drawbacks of a bespoke approach
+
+The maxim "if you aren't using a framework, you are building one" applies here.
```

Comment:
This is a really important point. Thinking about the style guide, I have some concern that a new frontend framework we build may end up languish in our maintenance backlog after a few years.